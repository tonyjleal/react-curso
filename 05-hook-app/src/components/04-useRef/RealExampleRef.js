import React, {useState} from 'react'
import '../02-useEffect/effects.css'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'


export const RealExampleRef = () => {

    const [show, setShow] = useState(false)

    const handleClick = () => {
        setShow(!show)
    }

    return (
        <div>
            <h1>RealExampleRef</h1>
            <hr />
            {
                show && 
                <MultipleCustomHooks />
            }

            <button 
                className='btn btn-primary'
                onClick={handleClick}>Show / Hide</button>
        </div>
    )
}
