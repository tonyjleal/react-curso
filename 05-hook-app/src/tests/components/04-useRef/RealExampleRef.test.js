import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"
import {shallow} from 'enzyme'

describe('test <RealExampleRef />', () => { 
    
    const wrapper = shallow(<RealExampleRef />)
    
    test('should show component success', () => { 
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
        
    })
    
    test('should show component <MultipleCustomHooks />', () => { 
        
        wrapper.find('button').simulate('click');

        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

    })


 })