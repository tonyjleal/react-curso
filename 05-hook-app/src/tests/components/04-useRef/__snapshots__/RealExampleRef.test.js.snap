// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <RealExampleRef /> should show component success 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</div>
`;
